test/e2e/framework/pv/wait.go

```diff
@@ -0,0 +1,141 @@
+/*
```

Comment:
@pohly implicitly asked whether these helper functions useful to others outside volume_modify test-suite. [See slack thread](https://kubernetes.slack.com/archives/C09QZ4DQB/p1722404575437579?thread_ts=1722370880.809189&cid=C09QZ4DQB)

I think a consumer like EBS CSI Driver would love the helper function `WaitForPersistentVolumeClaimModified` for our e2e tests instead of writing our own.

I'm fine with moving the other two helpers to volume_modify test-suite. Thoughts?

Reply:
I agree, these functions seem to be general enough to justify being here.

They look good to me from a SIG Testing perspective, too.

/approve
/hold

For second approval from SIG Storage.